### PR TITLE
Use a single CHANGELOG.md for all "subprojects"

### PR DESCRIPTION
**Pull request by martijnbastiaan**
We've been pretty bad at maintaining the various CHANGELOGs over the
past couple of years and for the past couple months we've only been
updating `clash-ghc/CHANGELOG.md`. @leonschoorl and I had an offline
discussion about it and figured the most practical solution was to just
keep a single log.

Symlinks are there for Hackage.

### CODE REVIEW

File: clash-prelude/CHANGELOG.md [hunk 1]
```diff
@@ -1,453 +0,0 @@
...
-  * Large overhaul of `TopEntity`  annotations
-  * PLL and other clock sources can now be instantiated using regular functions:
-    `Clash.Intel.ClockGen` and `Clash.Xilinx.ClockGen`.
-  * DDR registers:
-    * Generic/ASIC: `Clash.Explicit.DDR`
-    * Intel: `Clash.Intel.DDR`
-    * Xilinx: `Clash.Intel.Xilinx`
-  * `Bit` is now a `newtype` instead of a `type` synonym and will be mapped to
-    a HDL scalar instead of an array of one (e.g `std_logic` instead of
-    `std_logic_vector(0 downto 0)`)
```
Comment (leonschoorl) [inline clash-prelude/CHANGELOG.md]: Do we want to keep these?

Comment (martijnbastiaan) [reply clash-prelude/CHANGELOG.md]: It seemed "recent" enough that people might be interested in reading it.

Comment (martijnbastiaan) [reply clash-prelude/CHANGELOG.md]: Oh, I see what you're saying now. Yeah I guess.

Comment (martijnbastiaan) [reply clash-prelude/CHANGELOG.md]: Done.

**Pull request merged**